### PR TITLE
Added journald and cloudwatch callback plugins

### PR DESCRIPTION
**Pull request by chestm007**
##### SUMMARY
Introduced Journald and AWS cloudwatch_logs callback plugins

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAMES
callback_journald,
callback_cloudwatch_logs

##### ANSIBLE VERSION

```
ansible 2.6.2
  config file = None
  configured module search path = ['/home/max/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.5 (default, May 11 2018, 04:00:52) [GCC 8.1.0]

```

##### ADDITIONAL INFORMATION

expected output in journald logs
```
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] playbook started: test_playbook.yml
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] play started: test
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] task started: TASK: Gathering Facts
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] ['ok: ']: [localhost]
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] task started: TASK: ping
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] ['ok: ']: [localhost] => {'ping': 'pong', 'invocation': {'module_args': {'data': 'pong'}}, '_ansible_parsed': True, >
/usr/bin/ansible-playbook[11146]: max - PlaybookId[c8250a8c-cfe9-4599-8933-3e39fe8bbd51] COMPLETE: Playbook Succeeded. OK:2 CHANGED:0 FAILURES:0 UNREACHABLE:0
```

expected output in cloudwatch_logs
```
        {
            "timestamp": 1533269962192,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] playbook started: test_playbook.yml",
            "ingestionTime": 1533269964981
        },
        {
            "timestamp": 1533269962244,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] play started: test",
            "ingestionTime": 1533269964981
        },
        {
            "timestamp": 1533269962280,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] task started: TASK: Gathering Facts",
            "ingestionTime": 1533269964981
        },
        {
            "timestamp": 1533269964249,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] ['ok: ']: [localhost]",
            "ingestionTime": 1533269964981
        },
        {
            "timestamp": 1533269964267,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] task started: TASK: ping",
            "ingestionTime": 1533269964981
        },
        {
            "timestamp": 1533269964842,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] ['ok: ']: [localhost] => {'ping': 'pong', 'invocation': {'module_args': {'data': 'pong'}}, '_ansible_parsed': True, '_ansible_no_log': False, 'changed': False}",
            "ingestionTime": 1533269964981
        },
        {
            "timestamp": 1533269964846,
            "message": "max - PlaybookId[acf97c19-86ad-4ab2-8b11-1b4225e13b2e] COMPLETE: Playbook Succeeded. OK:2 CHANGED:0 FAILURES:0 UNREACHABLE:0",
            "ingestionTime": 1533269964981
        }

```

### CODE REVIEW

File: lib/ansible/plugins/callback/cloudwatch_logs.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    callback: cloudwatch_logs
+    callback_type: notification
+    requirements:
+      - whitelist in configuration
+      - boto3 (python library)
+    short_description: Sends play events AWS CloudWatch Logs
+    version_added: "2.5.4"
```
Comment (sivel) [inline lib/ansible/plugins/callback/cloudwatch_logs.py]: `version_added` should be `"2.7"`

Comment (chestm007) [reply lib/ansible/plugins/callback/cloudwatch_logs.py]: Have fixed this, thanks!

**Comment by chestm007** [comment]
Is there anything else i need to to with this?

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/94988/65/console

The failures are due to the following warnings:

```
02:50 + [WARNING]: Skipping plugin
02:50 +(/root/ansible/lib/ansible/plugins/callback/cloudwatch_logs.py) as it seems to
02:50 +be invalid: No module named boto3
02:50 + [WARNING]: Skipping plugin
02:50 +(/root/ansible/lib/ansible/plugins/callback/journald.py) as it seems to be
02:50 +invalid: No module named systemd.journal
```

File: lib/ansible/plugins/callback/cloudwatch_logs.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        env:
+          - name: CLOUDWATCH_LOGS_AWS_KMS_KEY_ID
+        ini:
+          - section: callback_cloudwatch_logs
+            key: aws_kms_key_id
+'''
+
+import getpass
+import os
+import boto3
```
Comment (mattclay) [inline lib/ansible/plugins/callback/cloudwatch_logs.py]: A try/except ImportError block is needed around all imports that are not from the Python standard library.

Take a look at other callback plugins to see how that is handled.

Comment (chestm007) [reply lib/ansible/plugins/callback/cloudwatch_logs.py]: done :)

**Comment by Akasurde** [comment]
Hi @chestm007, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**